go/adbc/driver/flightsql/record_reader.go

```diff
@@ -201,3 +208,23 @@ func (r *reader) Schema() *arrow.Schema {
 func (r *reader) Record() arrow.Record {
 	return r.rec
 }
+
+func removeSchemaMetadata(schema *arrow.Schema) *arrow.Schema {
+	fields := make([]arrow.Field, len(schema.Fields()))
+	for i, field := range schema.Fields() {
+		fields[i] = removeFieldMetadata(&field)
+	}
+	return arrow.NewSchema(fields, nil)
+}
+
+func removeFieldMetadata(field *arrow.Field) arrow.Field {
+	// XXX: this should recurse into the child fields, but there's not
+	// an easy way to navigate this generically - we can improve this
+	// upstream
+	return arrow.Field{
+		Name:     field.Name,
+		Type:     field.Type,
+		Nullable: field.Nullable,
+		Metadata: arrow.Metadata{},
+	}
```

Comment:
Could do:

```go
if n, ok := field.Type.(arrow.NestedType); ok {
    for i, f := range n.Fields() {
        // stuff
    }
}
```
?

Though upstream we should make an `EqualWithoutMetadata` for fields or something of the like....

Reply:
`// stuff` is the hard part because you can't generically reconstruct the type instance, right?

Reply:
(that said: not like you can in C++ or Java either.)

Reply:
I went ahead and implemented it.